File: source/entity/component/MaterialComponent.java
package entity.component;

import callback.EmptyCallback;
import entity.Entity;
import entity.abs.EntityComponent;
import gui.helper.GUIPopup;
import imgui.ImGui;
import material.Material;
import utility.Instance;

import java.io.Serializable;

public class MaterialComponent extends EntityComponent implements Serializable {
    public Material material = null;

    public MaterialComponent(Entity entity) {
        super(entity);
    }

    public void use(EmptyCallback callback) {
        if (Instance.isValid(material)) {
            material.use(callback);
        }
    }

    @Override
    public void renderGUI() {
        ImGui.bulletText(Instance.isValid(material) ? material.getName() : "None");

        GUIPopup.itemPopup("EditEntityMaterial", () -> {
            if (ImGui.button("Remove material")) {
                material = null;
                GUIPopup.close();
            }
        });
    }
}

File: source/entity/component/MaterialComponent.kt
package entity.component

import entity.Entity
import entity.abs.EntityComponent
import entity.material.Material
import gui.helper.GUIPopup
import imgui.ImGui
import java.io.Serializable

class MaterialComponent(entity: Entity) : EntityComponent(entity), Serializable {
    var material: Material? = null

    fun use(callback: () -> Unit) {
        material?.use(callback)
    }

    override fun renderGUI() {
        ImGui.bulletText(material?.name ?: "None")
        GUIPopup.itemPopup("EditEntityMaterial") {
            if (ImGui.button("Remove entity.material")) {
                material = null
                GUIPopup.close()
            }
        }
    }
}